system_status/system_status.py

```diff
@@ -4,93 +4,135 @@
+            email_data["medium2"],
+            email_data["low1"],
+            email_data["low2"],
+        )
+        sms_status = determine_status(
+            sms_data["high"],
+            sms_data["medium1"],
+            sms_data["medium2"],
+            sms_data["low1"],
+            sms_data["low2"],
+        )
+        logging.info("Email status: {}".format(email_status))
+        logging.info("SMS status: {}".format(sms_status))
     except Exception as e:
-        logging.error("System-status lambda error (determine_notification_status): {}".format(e))
-        return {
-            'statusCode': 500
-        }
-
-    # check api and admin response times
-    api_status = determine_site_status(URL_API, THRESHOLDS['api'])
-    admin_status = determine_site_status(URL_ADMIN, THRESHOLDS['admin'])
-    
-    # push statuses to s3 bucket
-    s3_client = boto3.client('s3')
+        logging.error(
+            "System-status lambda error (determine_notification_status): {}".format(e)
+        )
+        return {"statusCode": 500}
+
+    # Section 3: Check api and admin response times
+
+    api_status = determine_site_status(URL_API, THRESHOLDS["api"])
+    admin_status = determine_site_status(URL_ADMIN, THRESHOLDS["admin"])
+    logging.info("API status: {}".format(api_status))
+    logging.info("Admin status: {}".format(admin_status))
+
+    # Section 4: Push statuses to s3 bucket
+    s3_client = boto3.client("s3")
     bucket_name = BUCKET_NAME
-    object_key = 'response.json'
+    object_key = "response.json"
     status_data = [
-        {
-            'id': 'api',
-            'status': api_status
-        },
-        {
-            'id': 'admin',
-            'status': admin_status
-        },
-        {
-            'id': 'email',
-            'status': email_status
-        },
-        {
-            'id': 'sms',
-            'status': sms_status
-        }
+        {"id": "api", "status": api_status},
+        {"id": "admin", "status": admin_status},
+        {"id": "email", "status": email_status},
+        {"id": "sms", "status": sms_status},
     ]
     s3_client.put_object(
-        Body=json.dumps(status_data),
-        Bucket=bucket_name,
-        Key=object_key
+        Body=json.dumps(status_data), Bucket=bucket_name, Key=object_key
+    )
+
+    logging.info(
+        "Pushed status data to s3: {}".format(
+            {
+                "api": api_status,
+                "admin": admin_status,
+                "email": email_status,
+                "sms": sms_status,
+            }
+        )
     )
 
     return {
-        'body': {
-            'api': api_status,
-            'admin': admin_status,
-            'email': email_status,
-            'sms': sms_status
+        "body": {
+            "api": api_status,
+            "admin": admin_status,
+            "email": email_status,
+            "sms": sms_status,
         },
-        'statusCode': 200
+        "statusCode": 200,
     }
```

Comment:
Let's make sure we know when data fails to be uploaded to the S3 bucket.

```suggestion
    try :
        s3_client.put_object(
            Body=json.dumps(status_data), Bucket=bucket_name, Key=object_key
        )
    except botocore.exceptions.ClientError as e:
        logging.error(
            "System-status lambda error (pushing to s3): {}".format(e)
        )
    else:
        logging.info(
            "Pushed status data to s3: {}".format(
                {
                    "api": api_status,
                    "admin": admin_status,
                    "email": email_status,
                    "sms": sms_status,
                }
            )
        )
    finally:
        return {
            "body": {
                "api": api_status,
                "admin": admin_status,
                "email": email_status,
                "sms": sms_status,
            },
            "statusCode": 200,
        }
```

Reply:
fixed!